### PR TITLE
Feature/bi 8248 generalise appeal reasons object to integrate illness journey

### PR DESCRIPTION
**Pull request by mouhajer-ch**
### JIRA link
Resolves [BI-8248](https://companieshouse.atlassian.net/browse/BI-8248)

### Change description
Generalise appeal reasons object to integrate illness journey

### Work checklist

- [x] Tests added where applicable
- [ ] UI changes look good on mobile
- [ ] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/utils/appeal/extra.data.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import { Reasons } from 'app/models/Reasons';
+import { ReasonType } from 'app/models/fields/ReasonType';
+
+const getReasonType = (reasons: Reasons): ReasonType => {
+    return (ReasonType.illness in reasons && reasons.illness && !reasons.other)
+        ? ReasonType.illness
+        : ReasonType.other;
+};
+
+export const getReasonFromReasons = (reasons: Reasons): Illness | OtherReason | undefined => {
```
Comment (CH-JonathanBlake) [inline src/utils/appeal/extra.data.ts:13]: This is a pretty odd method name.

Comment (mouhajer-ch) [reply src/utils/appeal/extra.data.ts:13]: Thanks Joanathan, I am open to any suggestion :)

Comment (CH-JonathanBlake) [reply src/utils/appeal/extra.data.ts:13]: I still don't like this, however improved designs we brainstormed would require refactoring that is outside the scope of this change.

File: src/controllers/IllPersonController.ts [hunk 2]
```diff
@@ -62,30 +61,25 @@ export class IllPersonController extends BaseController<FormBody> {
     }
 
     protected prepareSessionModelPriorSave(appeal: Appeal, value: FormBody): Appeal {
-
+        // TBD: Remove OtherReason object when creating Illness if not Multiple reasons
```
Comment (CHdlewis) [inline src/controllers/IllPersonController.ts:64]: Is this comment still relevant

File: src/controllers/OtherReasonController.ts [hunk 3]
```diff
@@ -32,10 +32,11 @@ export class OtherReasonController extends SafeNavigationBaseController<OtherRea
     }
 
     protected prepareViewModelFromAppeal(appeal: Appeal): Record<string, any> & OtherReason {
-        return appeal.reasons?.other;
+        return appeal.reasons?.other as OtherReason;
     }
 
     protected prepareSessionModelPriorSave(appeal: Appeal, value: OtherReason): Appeal {
+        // TBD: Remove Illness object when creating OtherReason if not Multiple reasons
```
Comment (CHdlewis) [inline src/controllers/OtherReasonController.ts:39]: Is comment still necessary

**Review by CHdlewis — COMMENTED** [review]
Looks Good - Just keep an eye on those TBD comments.  If they can be ignored then I am happy to approve this.

File: src/controllers/IllPersonController.ts (hunk 2 above)
Comment (mouhajer-ch) [reply src/controllers/IllPersonController.ts:64]: Thanks Lewis, I think is important as a note for the team when the Illness journey object starts and how to behave.

File: src/controllers/OtherReasonController.ts (hunk 3 above)
Comment (mouhajer-ch) [reply src/controllers/OtherReasonController.ts:39]: Same but for OtherReason object

**Review by CHdlewis — APPROVED** [review]

**Pull request merged**